interfaces/ros1/wavemap_ros/src/ros_server.cc

```diff
@@ -0,0 +1,205 @@
+#include "wavemap_ros/ros_server.h"
+
+#include <std_srvs/Trigger.h>
+#include <wavemap/core/map/map_factory.h>
+#include <wavemap/io/file_conversions.h>
+#include <wavemap/pipeline/map_operations/map_operation_factory.h>
+#include <wavemap_msgs/FilePath.h>
+#include <wavemap_ros_conversions/config_conversions.h>
+
+#include "wavemap_ros/inputs/input_factory.h"
+#include "wavemap_ros/map_operations/map_ros_operation_factory.h"
+
+namespace wavemap {
+DECLARE_CONFIG_MEMBERS(RosServerConfig,
+                      (world_frame)
+                      (num_threads)
+                      (logging_level)
+                      (allow_reset_map_service));
+
+bool RosServerConfig::isValid(bool verbose) const {
+  bool all_valid = true;
+
+  all_valid &= IS_PARAM_NE(world_frame, std::string(""), verbose);
+  all_valid &= IS_PARAM_GT(num_threads, 0, verbose);
+
+  return all_valid;
+}
+
+// NOTE: If WavemapServerConfig::from(...) fails, accessing its value will throw
```

Comment:
WavemapServerConfig should be RosServer now

Reply:
Thanks!